editor/index.js

```diff
@@ -12,6 +19,10 @@ import './assets/stylesheets/main.scss';
 import Layout from './layout';
 import { createReduxStore } from './state';
 
+// Configure moment globally
+moment.locale( settings.l10n.locale );
```

Comment:
I don't think we necessarily need to solve this here. Proposed for more discussion on the related Trac ticket: https://core.trac.wordpress.org/ticket/39222#comment:35